### DISCUSSION TITLE
What does "Unchecked" mean?

### DISCUSSION THREAD

**Discussion by topbloc-beiswenger** [prisma/prisma#10121] [Q&A] [answered]
Hi there 👋

Most CRUD operations take in a `data` object with either a checked or unchecked type. For example, the [API reference for `create`](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#create) takes in `data` of type `XOR<UserCreateInput, UserUncheckedCreateInput>`. What exactly does unchecked type mean and how does Prisma choose whether it should use the checked or unchecked type?

Curious about this because I'm currently getting an error along the lines of `Unknown arg "teams" in data.teams for type UserUncheckedCreateInput` when trying to do:

```javascript
prisma.user.create({
    data: {
      teams: {
        connectOrCreate: teams.map((team) => {
          return {
            where: { name: team.name },
            create: {
              name: team.name,
              color: team.color,
            },
          };
        }),
      },
    },
  });
```

Thanks in advance 🙂

**Comment by austincrim** [comment, marked as answer]
Hi @topbloc-beiswenger! This is a great question.

"Unchecked" input types allow you to perform some operations that are often considered "dangerous", like directly writing foreign keys. Prisma allows you to choose either a safe or an unchecked input type when doing operations like `create`.

I'm unsure why you are hitting your specific error, but if you could share your schema file with me, I can do some digging on my side.

**Reply by topbloc-beiswenger** [reply]
Thanks for the response!

I found out that my problem was that `team.name` was undefined in `where: { name: team.name }` 😅

I think the error was just a bit misleading but I'll make sure to treat it as a catch-all in the future

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
